### ISSUE TITLE
Unexpected WRN_UninitializedNonNullableField warnings when ```this``` is assigned in constructor

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#30022]
```
        [Fact]
        public void StructAssignThis()
        {
            var source =
@"struct S
{
#pragma warning disable 0169
    private readonly string F;
    private string[] P { get; set; }
    internal S(S s)
    {
        this = s;
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition }, parseOptions: TestOptions.Regular8);
            comp.VerifyDiagnostics();
        }
```

Observed:
```
                // (6,14): warning CS8618: Non-nullable property 'P' is uninitialized.
                //     internal S(S s)
                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("property", "P").WithLocation(6, 14),
                // (6,14): warning CS8618: Non-nullable field 'F' is uninitialized.
                //     internal S(S s)
                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("field", "F").WithLocation(6, 14)
```
